strax/storage/s3.py

```diff
@@ -0,0 +1,217 @@
+pass the appropriate valeus to the constructor:
+
+  *  AWS_ACCESS_KEY_ID
+  *  AWS_SECRET_ACCESS_KEY
+
+"""
+
+import json
+import os
+import tempfile
+
+import boto3
+
+import strax
+from strax import StorageFrontend
+
+export, __all__ = strax.exporter()
+
+
+@export
+class SimpleS3Store(StorageFrontend):
+    """Frontend for S3 stores that just checks if data exists.
+
+    This uses boto3 for communicating, where you can look at their docs
+    to understand a lot of this.  S3 is an object store where each object is
+    chunk.  The bucket corresponds to strax key (run / plugin).
+
+    Currently, no run level metadata is stored.
+    """
+
+    def __init__(self,
+                 aws_access_key_id=None,
+                 aws_secret_access_key=None,
+                 endpoint_url='http://ceph-s3.mwt2.org',
+                 *args, **kwargs):
+        """
+        You must provide credentials to access your storage element.
+
+        :param aws_access_key_id: access key for S3-readable storage.
+        :param aws_secret_access_key: secret key for S3-readable storage.
+        :param endpoint_url: URL of S3-readable storage.
+
+        For other arguments, see DataRegistry base class.
+        """
+
+        super().__init__(*args, **kwargs)
+
+        # Get S3 protocol credentials
+        if aws_access_key_id is None:
+            if 'AWS_ACCESS_KEY_ID' not in os.environ:
+                raise EnvironmentError("S3 access key not specified")
+            aws_access_key_id = os.environ.get('AWS_ACCESS_KEY_ID')
+        if aws_secret_access_key is None:
+            if 'AWS_SECRET_ACCESS_KEY' not in os.environ:
+                raise EnvironmentError("S3 secret key not specified")
+            aws_secret_access_key = os.environ.get('AWS_SECRET_ACCESS_KEY')
+
+        #  Initialized connection to S3-protocol storage
+        self.s3client = boto3.client(aws_access_key_id=aws_access_key_id,
+                                     aws_secret_access_key=aws_secret_access_key,
+                                     endpoint_url=endpoint_url,
+                                     service_name='s3')
+
+        # Setup backends for reading
+        self.backends = [S3Backend(aws_access_key_id=aws_access_key_id,
+                                   aws_secret_access_key=aws_secret_access_key,
+                                   endpoint_url=endpoint_url)]
+
+    def _find(self, key, write, fuzzy_for, fuzzy_for_options):
+        """Determine if data exists
+
+        Search the S3 store to see if data is there.
+        """
+        # Don't support fuzzy matching since hard without RunDB
+        if fuzzy_for or fuzzy_for_options:
+            raise NotImplementedError("Can't do fuzzy with S3")
+
+        # Check exact match / write case
+        key_str = str(key)
+        bk = self.backend_key(key_str)
+
+        # Get list of all buckets / (run_id, lineage)
+        objects_list = self.s3client.list_buckets()
```

Comment:
I guess this is fine for now, but is there no way to access a bucket by name? The list of all buckets can get rather large. Guess we don't need this once we have a runs db though.

Reply:
You use `Bucket=self.key` later in the Saver's close, so apparently it is possible

Reply:
Exceptions in normal cases are pythonic (https://docs.quantifiedcode.com/python-anti-patterns/readability/asking_for_permission_instead_of_forgiveness_when_working_with_files.html). But in this case, at least according to https://stackoverflow.com/questions/26871884, it seems you get a generic ClientError which you then have to laboriously parse to ensure you're not catching other errors.

Anyway, it's fine, the run db should take care of locating data in the end.